### DISCUSSION TITLE
cargo build should build icx-asset

### DISCUSSION THREAD

**Discussion by jplevyak** [dfinity/sdk#2661] [Maintenance Tasks]
Do we want to build icx-asset by default from the root directory?  It isn't happening currently.

**Comment by viviveevee** [comment]
@smallstepman what do you think?

**Reply by jplevyak** [reply]
For context, my template for Rust projects is to include the asset canister functionality into a single Rust backend and then use icx-asset to sync the assets as part of the deploy workflow.  It is cheaper and easier to manage than having a separate asset canister.  As an aside it would be nice if icx-asset could use the default dfx identity for the pem file for this workflow, but I work around that with a makefile.
